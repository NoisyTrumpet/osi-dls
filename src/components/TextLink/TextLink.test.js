import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import {
  toBeInTheDocument,
  toHaveClass,
} from '@testing-library/jest-dom/matchers';
import '@testing-library/jest-dom/extend-expect';
import { TextLink } from './';

expect.extend({
  toBeInTheDocument,
  toHaveClass,
});

const URLToTest = 'https://www.hallmark.com/';

describe('<Textlink />', () => {
  test('Should render the component <TextLink /> with minimum required props (to, title)', () => {
    const { container } = render(<TextLink to="#" title="Amazing Text!" />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should find and click of the component <TextLink />, based on text-query, resulting in page load of https://www.hallmark.com/', () => {
    const { container, getByText } = render(
      <TextLink to={URLToTest} title="It renders a link!" />,
    );
    fireEvent.click(getByText('It renders a link!'));
    expect(document.querySelector('a').getAttribute('href')).toBe(URLToTest);
    expect(container).toMatchSnapshot();
  });

  test('Should render an icon within the component <TextLink />, with required props (to, title)', () => {
    const { container } = render(
      <TextLink
        icon="placeholder-bold"
        iconPos="right"
        to="#"
        title="We have a right-aligned placeholder icon"
      />,
    );
    expect(container.firstChild).toHaveClass('text-link has-icon icon-right');
    expect(container).toMatchSnapshot();
  });
});
