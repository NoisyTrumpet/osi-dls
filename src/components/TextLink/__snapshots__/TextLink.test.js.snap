// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textlink /> Should find and click of the component <TextLink />, based on text-query, resulting in page load of https://www.hallmark.com/ 1`] = `
<div>
  <a
    class="text-link"
    href="https://www.hallmark.com/"
  >
    <span
      class="typography typography-text-link"
    >
      It renders a link!
    </span>
  </a>
</div>
`;

exports[`<Textlink /> Should render an icon within the component <TextLink />, with required props (to, title) 1`] = `
<div>
  <a
    class="text-link has-icon icon-right"
    href="#"
  >
    <span
      class="typography typography-text-link"
    >
      We have a right-aligned placeholder icon
    </span>
    <span
      aria-hidden="true"
      class="icon icon-placeholder-bold icon-accentB icon-14"
      data-testid="icon"
    />
  </a>
</div>
`;

exports[`<Textlink /> Should render the component <TextLink /> with minimum required props (to, title) 1`] = `
<div>
  <a
    class="text-link"
    href="#"
  >
    <span
      class="typography typography-text-link"
    >
      Amazing Text!
    </span>
  </a>
</div>
`;
